src/ec/curve25519/ops.rs

```diff
@@ -39,6 +40,24 @@ impl ExtPoint {
         }
     }
 
+    pub fn from_bytes_vartime(bytes: &[u8; ELEM_LEN])
+                              -> Result<Self, error::Unspecified> {
+        let mut point = Self::new_at_infinity();
+
+        try!(bssl::map_result(unsafe {
+            GFp_x25519_ge_frombytes_vartime(&mut point, bytes)
+        }));
+
+        Ok(point)
+    }
+
+    pub fn to_bytes(&self) -> [u8; ELEM_LEN] {
```

Comment:
It looks like we _could_ move `self`. Grepping a bit, it seems we do that with similar methods in the `rsa::bigint` code. Since it is more idiomatic Rust to prefer refs to self, I would want to keep it as a ref unless we think it would prevent or discourage specific errors.

This seems like a gray area to me. On one hand, an owner of `self` calling this method multiple times needn't be incorrect per the meaning of the function on its own. In the actual usage, though, this would always be an error. This method is also private, so we don't have any unforeseen use cases to worry about, so the ref preference is less important.

I'm fine either way. If we do change it, we should change the name from `encode()` (which is what we ended up with) to `into_encoded()` or similar. What do you prefer?